### PR TITLE
Add type declaration for typescript user

### PR DESCRIPTION
**Pull request by NV4RE**
Also
- Fix eslint error on error.js file
- Bump ioredis to ^4.17.3

### CODE REVIEW

File: @types/redislock.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  /** delay   Time in milliseconds to wait between each attempt
+   * (default: 50 ms) */
+  delay: number;
+}
+
+/** Creates and returns a new Lock instance,
+ * configured for use with the supplied redis client, as well as options,
+ * if provided. The options object may contain following three keys,
+ * as outlined at the start of the documentation: timeout, retries and delay. */
+export function createLock(client: Redis, option: LockOption): Lock;
```
Comment (AVVS) [inline @types/redislock.d.ts]: should be Redis or Redis.Cluster

Comment (NV4RE) [reply @types/redislock.d.ts]: new Redis(); return Redis.Redis not Redis.Client and there are no exported Redis.Client

refs: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ioredis/index.d.ts#L34

Screenshot from 2020-10-28 19-59-12

Comment (AVVS) [reply @types/redislock.d.ts]: Cluster not client :)

Comment (NV4RE) [reply @types/redislock.d.ts]: Oh my bad, fixed.

**Comment by AVVS** [comment]
rewritten into typescript, will be published as `@microfleet/ioredis-lock`

**Pull request closed without merging**